lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1046,6 +1151,29 @@ SubExprInfo ExprEmitter::visitComb(MuxOp op) {
   return {Conditional, signedness};
 }
 
+SubExprInfo ExprEmitter::visitComb(StructExtractOp op) {
+  auto structPrec = emitSubExpr(op.input(), Selection);
+  os << '.' << op.field();
+  return {Selection, structPrec.signedness};
+}
+
+SubExprInfo ExprEmitter::visitComb(StructInjectOp op) {
+  StructType stype = op.getType().cast<StructType>();
+  os << "'{";
+  llvm::interleaveComma(stype.getElements(), os,
+                        [&](const StructType::FieldInfo &field) {
+                          os << field.name << ": ";
+                          if (field.name == op.field()) {
+                            emitSubExpr(op.newValue(), Selection);
+                          } else {
+                            emitSubExpr(op.input(), Selection);
+                            os << '.' << field.name;
```

Comment:
This isn't safe when op.input() is complicated (e.g. can cause exponential blowup of text).  Please change it to force the `op.input()` expression to be emitted out of line like rtl.sext does.